lib/vector-core/src/event/log_event.rs

```diff
@@ -293,10 +293,14 @@ impl LogEvent {
     }
 
     pub fn get_by_meaning(&self, meaning: impl AsRef<str>) -> Option<&Value> {
-        self.metadata()
-            .schema_definition()
-            .meaning_path(meaning.as_ref())
-            .and_then(|path| self.get(path))
+        if let Some(dropped) = self.metadata().dropped_field(&meaning) {
+            Some(dropped)
+        } else {
+            self.metadata()
+                .schema_definition()
+                .meaning_path(meaning.as_ref())
+                .and_then(|path| self.get(path))
```

Comment:
Is it possible for these two paths to get out of sync? E.g. if you drop a field with a given meaning and then subsequently write a new field and assign it that same meaning? Or is that not a problem because we're only using this in the `{only,except}_fields` path?

Reply:
If we do end up going with some kind of change like #18028 then it would be nice to undo this just to get rid of the oddity in the API and potential for mismatch. I'm not sure if that's just something like a TODO on the field or an issue, but maybe worth noting so that this doesn't stick around in the event that it becomes unneeded.